api/types/role.go

```diff
@@ -28,6 +28,7 @@ import (
 
 	"github.com/gogo/protobuf/proto"
 	"github.com/gravitational/trace"
+	"github.com/pborman/uuid"
```

Comment:
Is this package strictly necessary? If possible, try to limit the dependencies added in the api client.

Reply:
Having some kind of random value is important.  How about moving the downgrade logic to a standalone function in `services`?  Thats how we've dealt with limiting `api` imports elsewhere.

Reply:
I've moved this to lib/services/role.go in the latest commit to avoid the uuid dep in api/